src/main/java/com/artipie/front/settings/YamlCredentials.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.apache.commons.codec.binary.Hex;
+
+/**
+ * Yaml credentials parser.
+ * @since 1.0
+ */
+public final class YamlCredentials implements Credentials {
+
+    /**
+     * Yaml source.
+     */
+    private final YamlMapping mapping;
+
+    /**
+     * New yaml credentials.
+     * @param mapping Yaml
+     */
+    public YamlCredentials(final YamlMapping mapping) {
+        this.mapping = mapping;
+    }
+
+    @Override
+    public Optional<Credentials.User> user(final String name) {
+        return Optional.ofNullable(this.mapping.yamlMapping("credentials"))
+            .flatMap(cred -> Optional.ofNullable(cred.yamlMapping(name)))
+            .map(User::new);
+    }
+
+    /**
+     * Yaml user item.
+     * @since 1.0
+     */
+    private static final class User implements Credentials.User {
+
+        /**
+         * Yaml source.
+         */
+        private final YamlMapping mapping;
+
+        /**
+         * New user.
+         * @param mapping Yaml
+         */
+        private User(final YamlMapping mapping) {
+            this.mapping = mapping;
+        }
+
+        // @checkstyle ReturnCountCheck (30 lines)
+        @Override
+        @SuppressWarnings("PMD.OnlyOneReturn")
+        public boolean validatePassword(final String pass) {
+            final var config = this.mapping.string("pass");
+            if (config == null) {
+                throw new IllegalStateException(
+                    "invalid credentials configuration: `pass` field not found"
+                );
+            }
+            final var parts = config.split(":");
+            switch (parts[0]) {
+                case "plain":
+                    return Objects.equals(parts[1], pass);
+                case "sha256":
+                    return Objects.equals(parts[1], sha256(pass));
+                default:
+                    throw new IllegalStateException(
+                        String.format(
+                            "invalid credentials configuration: `pass` type `%s` is not supported",
+                                parts[0]
+                        )
+                    );
+            }
+        }
+
+        @Override
+        public Set<? extends String> groups() {
+            return Optional.ofNullable(this.mapping.yamlSequence("groups"))
+                .map(seq -> StreamSupport.stream(seq.spliterator(), false))
+                .orElse(Stream.empty())
+                .map(node -> node.asScalar().value())
+                .collect(Collectors.toSet());
+        }
+
+        /**
+         * SHA-256 hex of UTF-8 bytes of source string.
+         * @param src Source string
+         * @return Hex string
+         * @checkstyle MethodNameCheck (20 lines)
+         */
+        private static String sha256(final String src) {
```

Comment:
I'd suggest simplifying this method with `DigestUtils.sha256Hex(src)`, `DigestUtils` is a class from `org.apache.commons.codec`, we already have the dependency